src/main/java/com/bootcamp/demo/service/BookingService.java

```diff
@@ -0,0 +1,87 @@
+package com.bootcamp.demo.service;
+
+import com.bootcamp.demo.models.Booking;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.*;
+import com.google.firebase.cloud.FirestoreClient;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.logging.Logger;
+
+@Service
+public class BookingService implements IBookingService {
+    private static final Logger LOGGER = Logger.getLogger(BookingService.class.getName());
+    private static final String COLLECTION_PATH = "bookings";
+
+    private Firestore db;
+
+    public void setDb(Firestore db) {
+        this.db = db;
+    }
+
+    /**
+     * Retrieves a Booking entity by its id.
+     */
+    @Override
+    public Booking getBookingByID(final UUID bookingId) throws ExecutionException, InterruptedException, IllegalArgumentException {
+        if (bookingId == null) throw new IllegalArgumentException();
+        DocumentReference documentReference =  db.collection(COLLECTION_PATH)
+                .document(bookingId.toString());
+        ApiFuture<DocumentSnapshot> future = documentReference.get();
+        DocumentSnapshot document = future.get();
+        Booking booking;
+        if(document.exists()){
+            booking = document.toObject(Booking.class);
+            return booking;
+        }
+        return null;
+    }
+
+    /**
+     * Retrieves a Booking entity by its user's id.
+     */
+    @Override
+    public Set<Booking> getBookings(final UUID userId) throws ExecutionException, InterruptedException, IllegalArgumentException  {
+        if (userId == null) throw new IllegalArgumentException();

```

Comment:
Even when the body of `if` contains only one instruction, it is usually best to add surround it by { }. That is best practice as otherwise it is very easy to add another line here and result in unexpected behaviour.
Also, try to have one instruction/call per line, as it usually improves readability.